mmv1/third_party/terraform/services/compute/resource_compute_instance.go.tmpl

```diff
@@ -1553,11 +1576,20 @@ func resourceComputeInstanceCreate(d *schema.ResourceData, meta interface{}) err
 	}
 	{{- end }}
 
-	err = waitUntilInstanceHasDesiredStatus(config, d)
+	err = waitUntilInstanceHasDesiredStatus(config, d, "RUNNING")
 	if err != nil {
 		return fmt.Errorf("Error waiting for status: %s", err)
 	}
 
+	if val, ok := d.GetOk("desired_status"); ok {
+		if val.(string) != "RUNNING" {
+			err = changeInstanceStatusOnCreation(config, d, project, zone.Name, d.Get("desired_status").(string), userAgent)
```

Comment:
it all looks good only thing i would say is we can change `d.Get("desired_status").(string)` to be `val.(string)`

maybe even add a line that does `status := val.(string)`

Reply:
done